src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -347,11 +346,16 @@ def get_login_credentials(self, resource=None, client_id=None, subscription_id=N
 
         managed_identity_type, managed_identity_id = Profile._try_parse_msi_account_name(account)
 
-        # Cloud Shell is just a system assignment managed identity
         if in_cloud_console() and account[_USER_ENTITY].get(_CLOUD_SHELL_ID):
-            managed_identity_type = MsiAccountTypes.system_assigned
+            # Cloud Shell
+            from .auth.msal_authentication import CloudShellCredential
+            cred = CloudShellCredential()
```

Comment:
`CloudShellCredential` only exposes `get_token()`, so it doesn't work with Track 1 SDKs. Weather wrapping it with `CredentialAdaptor` works remains to be tested.